tests/testthat/test-tfb-wavelet.R

```diff
@@ -0,0 +1,192 @@
+library(purrr)
+library(wavethresh)
+library(checkmate)
+library(pillar)
+library(tidyverse)
+library(mgcv)
+library(memoise)
+
+set.seed(1234)
+data <- data.frame(id = rep(1:5, 8), 
+                   arg = rep(1:8, each = 5), 
+                   value = rnorm(8*5))
+
+temp <- tfb_wavelet.data.frame(data)
+tfb_wavelet.data.frame(data, filter.number = 8, levels = 1, type = "hard",
+                       policy = "universal")
+
+
+tf_evaluate(temp, c(1.5, 2.3, 4))
+
+is_tfb(temp)
+temp
+
+devtools::load_all(".")
+
+
+set.seed(1234)
+data <- data.frame(id = rep(1:5, 8), 
+                   arg = rep(1:8, each = 5), 
+                   value = rnorm(8*5))
+
+temp <- tfb_wavelet.data.frame(data, level = 2, filter_number = 3)
+tfb_wavelet.data.frame(data, filter.number = 8, levels = 1, type = "hard",
+                       policy = "universal")
+plot(temp)
+lines(tfd(temp), col = 2)
+
+
+
+# check constructors from tfd, matrix, data.frame, list
+context("tfb_wavelet constructor: basics")
+
+
+set.seed(1234)
+f <- function(t) 2*t + dbeta(t, 5, 7) * ifelse(t < .5, -1, 1) - .2 * dnorm(t, 0.2, 0.05)
+
+woo <- data.frame(id = rep(1:100, each = 256), 
+                  arg = rep(seq(0, 1, l = 256), 100))
+
+woo_df <- woo %>% mutate(data = f(arg) + rnorm(nrow(woo), sd = .5))
+woo_tfd <- tfd(woo_df)
+woo_mat <- as.matrix(woo_tfd)
+woo_tfb <- tfb_wavelet(woo_tfd, level = 2)
+
+test_that("tfb_wavelet defaults work for all kinds of regular input", {
+  for (dat in list(woo_df, woo_tfd)) {
+    woo_ <- try(tfb_wavelet(dat, verbose = FALSE))
+    expect_is(woo_, "tfb_wavelet")
+    expect_equal(length(woo_), length(woo_tfd))
+    expect_equivalent(tf_evaluations(woo_), tf_evaluations(woo_tfd),
+                      tolerance = 1)
+  }
+})
+
+
+test_that("tfb_wavelet works for different wavelet parameters", {
+  for (i in 1:10) {
+    for (j in 2:10) {
+      tfb_wavelet(woo_tfd, level = j, filter_number = i) 
+      print(paste0(i,".", j))
+    }
+  }
+})
+
+test_that("tfb_wavelet independent of datatype", {
+  expect_equal(tfb_wavelet(woo_tfd), tfb_wavelet(woo_df))
+  expect_equal(tfb_wavelet(woo_tfd), tfb_wavelet(woo_tfb))
```

Comment:
2nd test here fails, doesn't make sense because woo_tfb is already a heavily smoothed version of the original data
```r
plot(woo_tfb)
```
![image](https://user-images.githubusercontent.com/998541/69529025-b7f08080-0f6f-11ea-9107-24884543d03e.png)